onboard/cv/main.cpp

```diff
@@ -156,35 +221,40 @@ int main() {
     obstacleMessage.detected = false;
     #if OBSTACLE_DETECTION
       float pixelWidth = src.cols;
-      //float pixelHeight = src.rows;
       int roverPixWidth = calcRoverPix(distThreshold, pixelWidth);
-
+      
       /* obstacle detection */
       obstacle_return obstacle_detection =  avoid_obstacle_sliding_window(depth_img, src,  num_sliding_windows , roverPixWidth);
+      #if OBS_RECORD
+      vidWriteObs.write(src);
+      #endif
+
       if(obstacle_detection.bearing > 0.05 || obstacle_detection.bearing < -0.05) {
         // cout<< "bearing not zero!\n";
         obstacleMessage.detected = true;    //if an obstacle is detected in front
         obstacleMessage.distance = obstacle_detection.center_distance; //update LCM distance field
       }
       obstacleMessage.bearing = obstacle_detection.bearing;
 
-      #if PERCEPTION_DEBUG
-      // cout << "Turn " << obstacleMessage.bearing << ", detected " << (bool)obstacleMessage.detected<< endl;
-      #endif
-
     #endif
 
     lcm_.publish("/target_list", &arTagsMessage);
     lcm_.publish("/obstacle", &obstacleMessage);
 
     #if PERCEPTION_DEBUG
       imshow("depth", depth_img);
-      imshow("image", src);
+      imshow("Obstacle", src);
+      updateThresholds(thresh1,thresh2);
       waitKey(FRAME_WAITKEY);
     #endif
 
     j++;
   }
-
+  #if AR_RECORD
+  vidWrite.release();
+  #endif
+  #if OBS_RECORD
+  vidWriteObs.release();
```

Comment:
**NIT:** You may want to consider a wrapper around the VideoWriter objects that releases them upon destruction. This would remove the need for this block of release statements and RAII is generally a good practice to follow. That said I'm not super familiar with the Perception codebase so lmk if this wouldn't make sense

Reply:
^+1 to this. Having a wrapper to release the objects makes sense, and potentially we could also abstract away some of the logic related to the VideoWriter objects, which I think could make this file cleaner overall. Up to you though Matt, this could also be part of a later refactor if you think it doesn't make sense to change this now.

Reply:
I'll add it as a new member task for the future